### DISCUSSION TITLE
refactor code to use a new static field

### DISCUSSION THREAD

**Discussion by sebastian-toepfer** [google/error-prone#4221] [Q&A] [answered]
I have a project where I use Json-P massively. Now I have realized that I am using it in a very ineffective way. As described at https://github.com/jakartaee/jsonp-api/issues/154. Now I would like to replace all static method calls of class Json with calls against a static field of type JsonProvider. I like refaster, so I ask is it possible to do it with refaster?

let speak the code
i have:
```
class MassiveJson {

  JsonValue createJson() {
    return Json.createObjectBuilder().add("test", Json.createValue("test")).build();
  }
}
```

should be replaced by:
```
class MassiveJson {

  private static final JsonProvider JSONP = JsonProvider.provider();

  JsonValue createJson() {
    return JSONP.createObjectBuilder().add("test", JSONP.createValue("test")).build();
  }
}
```

**Comment by cushon** [comment, marked as answer]
Unfortunately his isn't something refaster is currently able to do, it doesn't have a way to add a new field declaration.

**Reply by sebastian-toepfer** [reply]
thx for carification. should we add this function? or it is a edge case which you will never support?

**Reply by cushon** [reply]
I wouldn't say _never_, but I think it's unlikely for refaster to support this any time soon. Refaster's main use-case is rewriting code inside a single method, and I don't see a clear path to adding support for higher level changes that need to add new declarations.
